src/flock_server/keybase.py

```diff
@@ -202,11 +202,62 @@ def _usage(self, cmd):
         )
 
     async def list_users(self, bot, event, args):
-        r = Search(index="user").query("match_all").execute()
-        users = [
-            "**{}** :point_right: **{}**".format(str(hit["username"]), str(hit["name"]))
-            for hit in r
-        ]
+        # Get all users
+        user_r = Search(index="user").query("match_all").execute()
+
+        # Start gathering data on users
+        users = {}
+        names = {}  # mapping name to username (host UUID)
+        for user_hit in user_r:
+            users[user_hit.username] = {"name": user_hit.name}
+            names[user_hit.name] = user_hit.username
```

Comment:
If it's the case that names aren't guaranteed to be unique, won't this overwrite the first instance of a name in the case that there is a duplicate? It will end up in the users hash, but the `for name in sorted(names.keys())` loop below will skip it and therefore it will never be pulled out and shown.

If so, a few ideas for fixing:
- Build one hash with the tuple `{user_hit.name, user_hit.username)` as the key, which should be unique and sort correctly. Then grab the first part of the tuple when you need the name below.
- Build one hash with the username as the key, and name in the value, and sort with `sorted(users.items(), key=lambda user: user[1]['name'])`.
- Almost certainly overkill, but we could create a UserRecord class with the attributes we want with `__gt__` and `__lt__ `functions. Then put a bunch of those in an array.

Reply:
That's a good point. I think I like your first idea